Dockerfile

```diff
@@ -0,0 +1,346 @@
+
+# print the JS build output
+RUN yarn config set enableInlineBuilds true
+
+#
+# STAGE: BUILD RS-DRIVE-ABCI
+#
+# This will prebuild majority of dependencies
+FROM sources AS build-drive-abci
+
+RUN mkdir /artifacts
+
+RUN --mount=type=cache,sharing=shared,target=/root/.cache/sccache \
+    --mount=type=cache,sharing=shared,target=${CARGO_HOME}/registry/index \
+    --mount=type=cache,sharing=shared,target=${CARGO_HOME}/registry/cache \
+    --mount=type=cache,sharing=shared,target=${CARGO_HOME}/git/db \
+    --mount=type=cache,sharing=shared,target=/platform/target \
+    export SCCACHE_SERVER_PORT=$((RANDOM+1025)) && \
+    if [[ -z "${SCCACHE_MEMCACHED}" ]] ; then unset SCCACHE_MEMCACHED ; fi ; \
+    cargo build \
+        --profile "$CARGO_BUILD_PROFILE" \
+        -p drive-abci \
+       --config net.git-fetch-with-cli=true && \
+    cp /platform/target/*/drive-abci /artifacts/drive-abci && \
+    sccache --show-stats
+
+#
+# STAGE: BUILD JAVASCRIPT INTERMEDIATE IMAGE
+#
+FROM sources AS build-js
+
+RUN mkdir /artifacts
+
+RUN --mount=type=cache,sharing=shared,target=/root/.cache/sccache \
+    --mount=type=cache,sharing=shared,target=${CARGO_HOME}/registry/index \
+    --mount=type=cache,sharing=shared,target=${CARGO_HOME}/registry/cache \
+    --mount=type=cache,sharing=shared,target=${CARGO_HOME}/git/db \
+    --mount=type=cache,sharing=shared,id=wasm_dpp_target,target=/platform/target \
+    --mount=type=cache,target=/tmp/unplugged \
+    cp -R /tmp/unplugged /platform/.yarn/ && \
+    export SCCACHE_SERVER_PORT=$((RANDOM+1025)) && \
+    if [[ -z "${SCCACHE_MEMCACHED}" ]] ; then unset SCCACHE_MEMCACHED ; fi ; \
+    export SKIP_GRPC_PROTO_BUILD=1 && \
+    yarn install && \
+    yarn build && \
+    cp -R /platform/.yarn/unplugged /tmp/ && \
+    sccache --show-stats
+
+# Remove Rust sources (not needed anymore)
+RUN rm -fr ./packages/rs-*
+
+#
+# STAGE: FINAL DRIVE-ABCI IMAGE
+#
+FROM alpine:${ALPINE_VERSION} AS drive-abci
+
+LABEL maintainer="Dash Developers <dev@dash.org>"
+LABEL description="Drive ABCI Rust"
+
+WORKDIR /var/lib/dash
+
+RUN apk add --no-cache libgcc libstdc++
+
+COPY --from=build-drive-abci /artifacts/drive-abci /usr/bin/drive-abci
+COPY --from=build-drive-abci /platform/packages/rs-drive-abci/.env.example /var/lib/dash/rs-drive-abci/.env
+
+# Double-check that we don't have missing deps
+RUN ldd /usr/bin/drive-abci
+
+# Create a volume
+VOLUME /var/lib/dash
+
+ENV DB_PATH=/var/lib/dash/rs-drive-abci/db
+
+#
+# Create new non-root user
+#
+ARG USERNAME=dash
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+RUN addgroup -g $USER_GID $USERNAME && \
+    adduser -D -u $USER_UID -G $USERNAME -h /var/lib/dash/rs-drive-abci $USERNAME && \
+    chown -R $USER_UID:$USER_GID /var/lib/dash/rs-drive-abci 
+
+USER $USERNAME
+
+ENV RUST_BACKTRACE=1
+WORKDIR /var/lib/dash/rs-drive-abci
+ENTRYPOINT ["/usr/bin/drive-abci"]
+CMD ["-vvvv", "start"]
+
+EXPOSE 26658
+
+#
+# STAGE: DASHMATE BUILD
+#
+FROM build-js AS build-dashmate
```

Comment:
I was a bit confused about what dashmate image we have since we don't dockerize dashmate. It took me some time to figure out that we have dashmate-helper subcomponent that we run together with dashmate services. What do you think @strophy @lklimek should we rename it to dashmate-helper?

Reply:
We discussed it on a call and agreed that it makes sense to rename